src/reactviews/pages/ExecutionPlan/executionPlanGraph.tsx

```diff
@@ -0,0 +1,203 @@
+	const [findNodeClicked, setFindNodeClicked] = useState(false);
+	const [findNodeOptions, setFindNodeOptions] = useState<string[]>([]);
+	const [highlightOpsClicked, setHighlightOpsClicked] = useState(false);
+	const LocalizedConstants = executionPlanState!.localizedConstants!;
+
+	useEffect(() => {
+		if (!executionPlanState || isExecutionPlanLoaded) return;
+
+		// @ts-ignore
+		window['mxLoadResources'] = false;
+		// @ts-ignore
+		window['mxForceIncludes'] = false;
+		// @ts-ignore
+		window['mxResourceExtension'] = '.txt';
+		// @ts-ignore
+		window['mxLoadStylesheets'] = false;
+		// @ts-ignore
+		window['mxBasePath'] = './src/reactviews/pages/ExecutionPlan/mxgraph';
+
+		const mxClient = azdataGraph.default();
+		// console.log(mxClient);
+		console.log("Execution Plan State: ", executionPlanState);
+
+		function loadExecutionPlan() {
+			if (executionPlanState && executionPlanState.executionPlanGraphs) {
+				const executionPlanRootNode = executionPlanState.executionPlanGraphs[graphIndex].root;
+				const executionPlanView = new ExecutionPlanView(executionPlanRootNode);
+				const executionPlanGraph = executionPlanView.populate(executionPlanRootNode);
+				console.log("Graph: ", executionPlanGraph);
+
+				const div = document.getElementById(`queryPlanParent${graphIndex + 1}`);
+				// create a div to hold the graph
+				const queryPlanConfiguration = {
+					container: div,
+					queryPlanGraph: executionPlanGraph,
+					iconPaths: utils.getIconPaths(),
+					badgeIconPaths: utils.getBadgePaths(),
+					expandCollapsePaths: utils.getCollapseExpandPaths(),
+					showTooltipOnClick: true
+				};
+				const pen = new mxClient.azdataQueryPlan(queryPlanConfiguration);
+				pen.setTextFontColor('var(--vscode-editor-foreground)'); // set text color
+				pen.setEdgeColor('var(--vscode-editor-foreground)'); // set edge color
+
+				executionPlanView.setDiagram(pen);
+				executionPlanView.zoomToFit();
+
+				setExecutionPlanView(executionPlanView);
+				setIsExecutionPlanLoaded(true);
+				setFindNodeOptions(executionPlanView.getUniqueElementProperties());
+				setZoomNumber(executionPlanView.getZoomLevel());
+
+				let tempQuery = executionPlanState.executionPlanGraphs[graphIndex].query
+				if (graphIndex != 0) {
+					const firstAlphaIndex = tempQuery.search(/[a-zA-Z]/);
+
+					if (firstAlphaIndex !== -1) {
+						tempQuery = tempQuery.slice(firstAlphaIndex);
+					}
+				}
+				setQuery(tempQuery);
+				setCost(executionPlanView.getTotalRelativeCost());
+			}
+			else {
+				return;
+			}
+		}
+		loadExecutionPlan();
+
+	}, [executionPlanState]);
+
+	const handleCustomZoomInput = async () => {
+		if (executionPlanView) {
+			executionPlanView.setZoomLevel(zoomNumber);
+			setExecutionPlanView(executionPlanView);
+			setZoomNumber(executionPlanView.getZoomLevel());
+		}
+		setCustomZoomClicked(false);
+	};
+
+	const getQueryCostPercentage = () => {
+		const percentage = (cost / executionPlanState!.totalCost!) * 100;

```

Comment:
Is this every a divide by zero situation?

Reply:
Not really. it's only zero if there are no graphs returned by the api call, but one of the prereq conditions of the executionPlanGraph component even showing up is that there is a graph for it to show up.

Reply:
It looks like the total cost can be 0 with a query such as "SELECT 1", which causes the percent to display as NaN.

![image](https://github.com/user-attachments/assets/7f1d9ebf-5de3-424a-a64b-b94c6da943c1)